src/components/Card.js

```diff
@@ -1,52 +1,85 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardTemplate,
+    handleImageClick,
+    handleTrashClick,
+    api
+  ) {
     this._name = name;
     this._link = link;
-    this._cardSelector = cardSelector;
+    this._id = _id;
+    this.isLiked = isLiked;
+    this._cardTemplate = cardTemplate;
     this._handleImageClick = handleImageClick;
-    this._element = this._getTemplate();
-    this._likeButton = this._element.querySelector(".card__like-button");
-    this._deleteButton = this._element.querySelector(".card__delete-button");
-    this._cardImage = this._element.querySelector(".card__image");
-    this._cardTitle = this._element.querySelector(".card__title");
+    this._handleTrashClick = handleTrashClick;
+    this._api = api;
+    this.element = this._getTemplate();
+    this.likeButton = this.element.querySelector(".card__like-button");
+    this._deleteButton = this.element.querySelector(".card__delete-button");
+    this._cardImage = this.element.querySelector(".card__image");
+    this._cardTitle = this.element.querySelector(".card__title");
+    this._deleteCardModal = document.querySelector("#delete-card-modal");
     this._setEventListeners();
   }
 
   _getTemplate() {
-    const cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
+    const cardElement = this._cardTemplate.content
+      .querySelector(".card")
       .cloneNode(true);
     return cardElement;
   }
 
   _setEventListeners() {
-    this._likeButton.addEventListener("click", () => {
-      this._handleCardLike();
+    this.likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this);
     });
 
     this._deleteButton.addEventListener("click", () => {
-      this._handleCardDelete();
+      this._handleTrashClick(this);
     });
 
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick({ name: this._name, link: this._link });
     });
   }
 
-  _handleCardLike() {
-    this._likeButton.classList.toggle("card__like-button_active");
+  _handleLikeClick() {
+    if (this.isLiked) {
+      this._api
+        .unlikeCard(this._id)
+        .then(() => {
+          this.isLiked = false;
+          this._updateLikeButton();
+        })
+        .catch((error) => console.error("Error unliking card:", error));
+    } else {
+      this._api
+        .likeCard(this._id)
+        .then(() => {
+          this.isLiked = true;
+          this._updateLikeButton();
+        })
+        .catch((error) => console.error("Error liking card:", error));
+    }
   }
 
-  _handleCardDelete() {
-    this._element.remove();
-    this._element = null;
+  _handleTrashClick() {
```

Comment:
Please, delete it from `Card.js` because any modal should be created in `index.js` using a class extended from `Popup`. There are `open` and `_handleEscClose` methods for that


`_handleTrashClick` is not needed here because you reassign it in the constructor in line 15